core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -419,4 +422,33 @@ public void testCanReadOldCompressedManifestFiles() throws Exception {
     List<FileScanTask> tasks = Lists.newArrayList(reloaded.newScan().planFiles());
     Assert.assertEquals("Should scan 1 files", 1, tasks.size());
   }
+
+  @Test
+  public void testConcurrentAppend() throws Exception {
+    assertTrue("Should create v1 metadata",
+            version(1).exists() && version(1).isFile());
+    File dir = temp.newFolder();
+    dir.delete();
+
+    Table tableWithHighRetries = TABLES.create(SCHEMA, SPEC, new HashMap<String, String>() {
+      {
+        put(TableProperties.COMMIT_NUM_RETRIES, "1000");
+      }
+    }, dir.toURI().toString());
+    int threadsCount = 30;
+    Thread[] threads = new Thread[threadsCount];
+    for (int i = 0; i < threadsCount; i++) {
+      threads[i] = new Thread(() -> tableWithHighRetries.newAppend().appendFile(FILE_A).commit());
+      threads[i].start();
+    }
+    Arrays.stream(threads).forEach(t -> {
+      try {
+        t.join();
+      } catch (InterruptedException e) {
+        // intentionally swallow to check result later
+      }
+    });
+    tableWithHighRetries.refresh();
+    assertEquals(threadsCount, Lists.newArrayList(tableWithHighRetries.snapshots()).size());
```

Comment:
Is there no danger here of having a potentially flakey test? I see that the commit number of retries is set to 1000.

But our CI environment can get very busy. I worry that having to set the retries to 1000 will cause the amount of time CI takes to increase by quite a bit. We do share test resources with the entirety of the ASF after all.